x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -541,7 +541,9 @@ private void createApiKeyAndIndexIt(
     ) {
         final Instant created = clock.instant();
         final Instant expiration = getApiKeyExpiration(created, request.getExpiration());
-        final SecureString apiKey = UUIDs.randomBase64UUIDSecureString();
+        // the difference between 16 and 18 effectively results in the same "encoded" API Key that's sent in HTTP request headers,
+        // dues to base64 padding
+        final SecureString apiKey = getBase64SecureRandomString(request.getType() == ApiKey.Type.CROSS_CLUSTER ? 16 : 18);
         assert ApiKey.Type.CROSS_CLUSTER != request.getType() || API_KEY_SECRET_LENGTH == apiKey.length()
             : "Invalid API key (name=[" + request.getName() + "], type=[" + request.getType() + "], length=[" + apiKey.length() + "])";
```

Comment:
As @n1v0lg pointed out, we'd need to support the current length for cross cluster API keys.
Maybe other (external, i.e. non-ES) consumers rely on the "api_key" length not changing, but hopefully they use the "encoded" one in the API response, which doesn't change.

Reply:
e.g.:
```
{
  "id" : "bZ0Y7JEBuvKKBRFbe4xD",
  "name" : "test",
  "api_key" : "BGEQu0kTIU8x7b63k9Np41dE",
  "encoded" : "YlowWTdKRUJ1dktLQlJGYmU0eEQ6QkdFUXUwa1RJVTh4N2I2M2s5TnA0MWRF"
}
```
The `encoded` is still `60` chars, but `api_key` changes from `22` to `24`.